earth2studio/data/utils.py

```diff
@@ -75,34 +85,103 @@ def fetch_data(
         da0 = da0.assign_coords(time=time)
         da.append(da0)
 
-    return prep_data_array(xr.concat(da, "lead_time"), device=device)
+    return prep_data_array(
+        xr.concat(da, "lead_time"),
+        device=device,
+        interp_to=interp_to,
+        interp_method=interp_method,
+    )
 
 
 def prep_data_array(
     da: xr.DataArray,
     device: torch.device = "cpu",
+    interp_to: CoordSystem = None,
+    interp_method: str = "nearest",
 ) -> tuple[torch.Tensor, CoordSystem]:
     """Prepares a data array from a data source for inference workflows by converting
     the data array to a torch tensor and the coordinate system to an OrderedDict.
 
+    If desired, xarray interpolation/regridding in the spatial domain can be used
+    by passing a target coordinate system via the optional `interp_to` argument.
+
     Parameters
     ----------
     da : xr.DataArray
         Input data array
     device : torch.device, optional
         Torch devive to load data tensor to, by default "cpu"
+    interp_to : CoordSystem, optional
+        If provided, the fetched data will be interpolated to the coordinates
+        specified by lat/lon arrays in this CoordSystem
+    interp_method : str
+        Interpolation method to use with xarray (by default 'nearest')
 
     Returns
     -------
     tuple[torch.Tensor, CoordSystem]
         Tuple containing output tensor and coordinate OrderedDict
     """
 
-    out = torch.Tensor(da.values).to(device)
-
+    # Initialize the output CoordSystem
     out_coords = OrderedDict()
     for dim in da.coords.dims:
-        out_coords[dim] = np.array(da.coords[dim])
+        if dim in ["time", "lead_time", "variable"]:
+            out_coords[dim] = np.array(da.coords[dim])
+
+    # Fetch data and regrid if necessary
+    if interp_to is not None:
```

Comment:
Just noting that we need to figure out how to deal with interpolation better this now makes three spots we have such functionality: map_coords, interp.py and now here.